### ISSUE TITLE
[BUG]Proform Elliptical Bike Hybrid does not transmit resistance (identifier "I_EL") due to possibly unflushed class?

### ISSUE THREAD

**Issue by twilightravengames** [cagnulein/qdomyos-zwift#1067]
Proform Hybrid Elliptical is identified as of class nordictrackelliptical
by identifier "I_EL" on line 776 of bluetooth.cpp.

Resistance is not modulated in QZ App according to changes on interface panel of proform equipment mentioned
resistance is not transmitted via bridge mechanism from proform equipment mentioned to VzFit Oculus Android OS app

Class is constructed as:
nordictrackelliptical(noWriteResistance, noHeartService,
                                                                  bikeResistanceOffset, bikeResistanceGain);
It appears from observation that bikeResistanceOffset and bikeResistanceGain are not being modulated by QZ interface to bike.
Unable to find class definition (headeer for nordictrackcelliptical) in solution provided by repository. 

Code needs to be strongly refactored and extracted. No offense meant, but no program should have that long of an if-else chain without being refactored and extracted according to common coding practices and industry standards.

**Comment by twilightravengames** [comment]
I can relate to doing everything by yourself, and I don't have a shippable
product yet since you mention it. I want to build a metaverse standard off
of the current web infrastructure (it doesn't use a private walled garden
approach like most companies). I know how to do it technically,, but I am
having trouble finding time, money, and resources. Especially when I work
full time for a CGI company.

I can help with the refactoring of your code if you would like. i'll see
what i can do about the debug log. this is mainly for my mother, Betty
MIller, who purchased your app and asks me questions about it non-stop.

Thanks.

Ted

**Comment by drmason789** [comment]
@twilightravengames concerning refactoring...I've also made this observation about the lengthy device detection if statement, and I've had to bypass it for my work adding a USB/Serial device #855.

I have some refactoring work underway, although in the last months I've been preoccupied with my employment.
- Regarding the enormous if statement, #887 I've moved the blocks into separate functions in the bluetooth class here: https://github.com/drmason789/qdomyos-zwift/blob/bluetooth_polymorphism_887/src/bluetooth.cpp . This PR is getting a bit out of date. and I have another uncommitted implementation that uses a list of functors rather than a list of function calls. 
- #994 #995 Encapsulating the storage and cleanup of the virtual device. 
- #996 and #889 will address problems with the device detection if statement I've encountered while adding support for a non-bluetooth device

However, with unexpected changes in functionality potentially causing @cagnulein a flood of 100s of support emails, this is on hold while I implement unit testing initially for the device detection: #911. I probably won't be submitting a PR for #911 until early January 2023.
